llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -308,8 +308,11 @@ AMDGPUTargetLowering::AMDGPUTargetLowering(const TargetMachine &TM,
   setTruncStoreAction(MVT::v2f64, MVT::v2f32, Expand);
   setTruncStoreAction(MVT::v2f64, MVT::v2f16, Expand);
 
+  setTruncStoreAction(MVT::v3i32, MVT::v3i8, Expand);
```

Comment:
I suspect we're missing test coverage (or we're doing unnecessary work) for the load half of this

Reply:
@arsenm - may I merge the patch without taking care of this?

Reply:
yes